security/security-spring/src/main/java/org/cbioportal/security/spring/CancerStudyPermissionEvaluator.java

```diff
@@ -181,6 +182,15 @@ public boolean hasPermission(Authentication authentication, Serializable targetI
                 }
             }
             return true;
+        } else if ("List<SampleListId>".equals(targetType)) {
```

Comment:
I should make clear that the above loop is not for a PostFilter, it is for when you are checking the permission of a parameter that is a list.  So it will be when a person has submitted a list of IDs and they do not have permission to access one or more of the elements.  If we are filtering we should always use the other hasPermission that is checking individual objects, not lists.